lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2918,101 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
   return Result;
 }
 
+void Parser::SkipInvalidExprAndParseRelativeBoundsClause(ExprResult &Result,
```

Comment:
Please remove this function from this pull request.   I think improving error recovery for invalid bounds expressions should be in a separate pull request.